_posts/2017-08-16-Refactoring-with-ReactJS-testing.md

```diff
@@ -0,0 +1,253 @@
+  import FailureBanner from '../../../jsx/failureBanner.jsx';
+{% endhighlight %}
+Importing tested component from its relative path. Here as everywhere you might notice that we are importing components with their JSX extensions. This is a shortcoming in our build configuration that we were not yet able to overcome. Presently if you omit JSX it will get resolved to JS and fail to find the component.
+<p />Following `describe` method is comes from Mocha framework to wrap a suit of tests.<p />
+Let's look inside `beforeEach` call and dive into a bit more of a detail on rendering the component for testing.
+{% highlight javascript linenos %}
+beforeEach(function(done){
+        component = renderComponent(FailureBanner, {message: 'test this failure message', isShown: true});
+        done();
+    });
+{% endhighlight %}
+As we set out to render a component using our `renderComponent` helper method we have to pass in the ReactJS component being renderd as well as an object that reporesents properties we would like for this object to assume prior to rendering. As you can see the object is going to be shown with the included message.
+It is important to include `done` parameter into the function call and then to call it. We need to ensure that rendering of the object was completed before any tests have been performed on the rendered object. As you can see done parameter is a function that needs to be called to indicate that processing has been completed and the rest of the test can be executed.
+<p />
+Now let's take a look back at the component for reference in order to understand the assertion line.
+{% highlight javascript linenos %}
+import React from 'react';
+import { render } from 'react-dom';
+
+export default class FailureBanner extends React.Component {
+
+    render(){
+        if(this.props.isShown) {
+            return (
+                <div className="flamingo " data-test-component="failureBanner">
+                    <div role="alert" className="alert alert-danger" id="msg-text">
+                        <div className="alert-left flex-center">
+                            <i className="fa fa-check fa-lg" aria-hidden="true"></i>
+                        </div>
+                        <p id="msg-message">Stuff here:{this.props.message}</p>
+                    </div>
+                </div>
+            );
+        }
+        return null;
+    }
+}
+{% endhighlight %}
+At the line `9` you can see `data-test-component="failureBanner"` and when this component is renderd this is the tag that we are latching on in this test.
+{% highlight javascript %}
+  expect(component.$el.data('test-component')).to.equal('failureBanner');
+{% endhighlight %}
+Now its time to look at the helper methods that we used here.
+
+# Test Helper Methods
+We were fortunate enough to happen upon an excellent [React/Redux](https://www.udemy.com/react-redux-tutorial/) course on Udemy tought by [Stephen Grider](https://github.com/StephenGrider).
```

Comment:
I think it would be helpful to mention that the specific advanced course you're linking includes testing as well as other topics.